test/units/utils/collection_loader/test_collection_loader.py

```diff
@@ -37,7 +38,11 @@ def test_find_module_py3():
     # setuptools may fall back to find_module on Python 3 if find_spec returns None
     # see https://github.com/pypa/setuptools/pull/2918
     assert path_hook_finder.find_spec('missing') is None
-    assert path_hook_finder.find_module('missing') is None
+    if sys.version_info < (3, 12):
```

Comment:
Let's invert this to dedent stuff below:
```suggestion
    if sys.version_info >= (3, 12):
        return
```

Reply:
If this assert ends up in a separate test (see my previous comment), the the version check can also be done with a decorator.